pkg/netpol/connlist/connlist_test.go

```diff
@@ -1,469 +1,559 @@
 package connlist
 
 import (
-	"errors"
-	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
 
 	"github.com/np-guard/netpol-analyzer/pkg/netpol/common"
 	"github.com/np-guard/netpol-analyzer/pkg/netpol/internal/testutils"
-	"github.com/np-guard/netpol-analyzer/pkg/netpol/scan"
 
 	"github.com/stretchr/testify/require"
 )
 
-func getConnlistFromDirPathRes(stopOnErr bool, path string) (*ConnlistAnalyzer, []Peer2PeerConnection, error) {
-	var analyzer *ConnlistAnalyzer
-	if stopOnErr {
-		analyzer = NewConnlistAnalyzer(WithStopOnError())
-	} else {
-		analyzer = NewConnlistAnalyzer()
-	}
+const connlistExpectedOutputFileNamePrefix = "connlist_output."
+const underscore = "_"
 
-	res, _, err := analyzer.ConnlistFromDirPath(path)
-	return analyzer, res, err
+var allFormats = []string{common.TextFormat, common.JSONFormat, common.CSVFormat, common.MDFormat, common.DOTFormat}
+
+// helping func - returns test's dir path from test's dir name
+func getDirPathFromDirName(dirName string) string {
+	return filepath.Join(testutils.GetTestsDir(), dirName)
 }
 
-type testEntry struct {
-	testDirName          string
-	outputFormats        []string
-	generateActualOutput bool // if true, overrides existing expected output file
+// helping func - creates ConnlistAnalyzer with desired opts and returns the analyzer with connlist from provided directory
+func getConnlistFromDirPathRes(opts []ConnlistAnalyzerOption, dirName string) (*ConnlistAnalyzer, []Peer2PeerConnection, error) {
+	analyzer := NewConnlistAnalyzer(opts...)
+	res, _, err := analyzer.ConnlistFromDirPath(getDirPathFromDirName(dirName))
+	return analyzer, res, err
 }
 
-const expectedOutputFileNamePrefix = "connlist_output."
+// helping func - creates the analyzer , gets connlist and writes it to string and verifies results
+func verifyConnlistAnalyzeOutputVsExpectedOutput(t *testing.T, analyzerOptions []ConnlistAnalyzerOption, dirName,
```

Comment:
Since this function can be called from a loop that runs a single test dir with various output formats, the message of a failed test should also include the output format  it was run with (in addition to the test name) , so that one can debug a failure more easily.